tests/failuremode.rs

```diff
@@ -0,0 +1,168 @@
+use crate::util::common::wasm_module;
+use proxy_wasm_test_framework::tester;
+use proxy_wasm_test_framework::types::{Action, BufferType, LogLevel, MapType, ReturnType};
+use serial_test::serial;
+
+pub mod util;
+
+#[test]
+#[serial]
+fn it_runs_next_action_on_failure_when_failuremode_is_allow() {
+    let args = tester::MockSettings {
+        wasm_path: wasm_module(),
+        quiet: false,
+        allow_unexpected: false,
+    };
+    let mut module = tester::mock(args).unwrap();
+
+    module
+        .call_start()
+        .execute_and_expect(ReturnType::None)
+        .unwrap();
+
+    let root_context = 1;
+    let cfg = r#"{
+        "services": {
+            "limitador": {
+                "type": "ratelimit",
+                "endpoint": "limitador-cluster",
+                "failureMode": "deny",
+                "timeout": "5s"
+            },
+            "limitador-unreachable": {
+                "type": "ratelimit",
+                "endpoint": "unreachable-cluster",
+                "failureMode": "allow",
+                "timeout": "5s"
+            }
+        },
+        "actionSets": [
+        {
+            "name": "some-name",
+            "routeRuleConditions": {
+                "hostnames": ["example.com"]
+            },
+            "actions": [
+            {
+                "service": "limitador-unreachable",
+                "scope": "a",
+                "data": [
+                    {
+                        "expression": {
+                            "key": "l",
+                            "value": "1"
+                        }
+                    }
+                ]
+            },
+            {
+                "service": "limitador",
+                "scope": "a",
+                "data": [
+                    {
+                        "expression": {
+                            "key": "l",
+                            "value": "1"
+                        }
+                    }
+                ]
+            }]
+        }]
+    }"#;
+
+    module
+        .call_proxy_on_context_create(root_context, 0)
+        .expect_log(Some(LogLevel::Info), Some("#1 set_root_context"))
+        .execute_and_expect(ReturnType::None)
+        .unwrap();
+    module
+        .call_proxy_on_configure(root_context, 0)
+        .expect_log(Some(LogLevel::Info), Some("#1 on_configure"))
+        .expect_get_buffer_bytes(Some(BufferType::PluginConfiguration))
+        .returning(Some(cfg.as_bytes()))
+        .expect_log(Some(LogLevel::Info), None)
+        .execute_and_expect(ReturnType::Bool(true))
+        .unwrap();
+
+    let http_context = 2;
+    module
+        .call_proxy_on_context_create(http_context, root_context)
+        .expect_log(Some(LogLevel::Debug), Some("#2 create_http_context"))
+        .execute_and_expect(ReturnType::None)
+        .unwrap();
+
+    let first_call_token_id = 42;
```

Comment:
Should we just leave this out and have it hard-coded the same as the second token? or use a diff token than 42 for both?